### PR TITLE
Add Szalinski eval; parameterize match limits

### PR DESCRIPTION
**Pull request by rtjoa**

### CODE REVIEW

File: src/enumo/ruleset.rs [hunk 1]
```diff
@@ -491,7 +491,7 @@ impl<L: SynthLanguage> Ruleset<L> {
             }
         }
 
-        let out_egraph = scheduler.run_derive(&egraph, self, rule);
+        let out_egraph = scheduler.run_internal(&egraph, self, Some(rule));
```
Comment (ajpal) [inline src/enumo/ruleset.rs:494]: wait, let's not get rid of `run_derive`, we still want that

Comment (rtjoa) [reply src/enumo/ruleset.rs:494]: `run_derive` use to just call `run_internal` with a match limit of 1000. Now, the match limit is in the scheduler.

I'll leave `run_derive` in, and have it just wrap `rule` in `Some`?

Comment (ajpal) [reply src/enumo/ruleset.rs:494]: yep

**Review by mwillsey — APPROVED** [review]
Looks good to me! (note: I only reviewed the Sz stuff)

**Review by chandrakananandi — APPROVED** [review]
Nice work!

**Comment by rtjoa** [comment]
@ajpal Am I good to merge?

**Pull request merged**